distributed/client.py

```diff
@@ -394,6 +396,16 @@ def __init__(self, address=None, start=True, loop=None, timeout=5,
         self.connection_args = self.security.get_connection_args('client')
         self._connecting_to_scheduler = False
 
+        if loop is None:
+            self._should_close_loop = None
+            if in_ioloop():
+                self.loop = IOLoop.current()
```

Comment:
How about:
```
self.loop = IOLoop.current(instance=False) or IOLoop()
```
(`IOLOoop.current(instance=False)` returns None if there is no registered IOLoop for this thread)